src/material-experimental/mdc-menu/harness/mdc-menu-harness.ts

```diff
@@ -40,7 +40,7 @@ export class MatMenuHarness extends ComponentHarness {
   }
 
   async isOpen(): Promise<boolean> {
-    throw Error('not implemented');
+    return Promise.reject(Error('not implemented'));
```

Comment:
Does the lint check require us to do it this way? It's kind of nice just doing `throw`, it's a lot more concise

Reply:
It requires there to be either a `return` or an `await`.

Reply:
Should we just not use the `no-async-without-await` rule`. IIRC such methods are marked as `async` for public API compatibility, so I don't think there would be anything wrong with having an `async` function not use `await` (in terms of method/function signature)

Reply:
Well the expectation is that an `async` method does return a `Promise` this makes sure that we're consistent.

Reply:
Well if it throws, it doesn't matter what value is returned. Even if the function has no body, a promise will be returned if the `async` keyword is used. So in this scenario, I don't think semantically it makes sense to expect a return statement if the method is _always_ aborted with an exception.

Reply:
I think that these methods that just throw are kinda rare anyway. This also caught a few cases where we legitimately were using `async` when we didn't need to.

Reply:
Yeah, it's a trade-off. I don't feel too strong about it. Just want to make it clear that the rule is *definitely* not perfect and this is a false-positive rule failure.

Reply:
I don't see much benefit from the `no-async-without-await` rule. I don't like how it forces us to put empty return statements either. I vote for not using it

Reply:
Isn't this basically the same as the `noImplicitReturns` compiler option that we have turned on already?